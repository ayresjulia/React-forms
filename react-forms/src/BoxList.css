.BoxList {
	display: flex;
	flex-direction: row;
}

.BoxList-form {
	display: inline-block;
}
