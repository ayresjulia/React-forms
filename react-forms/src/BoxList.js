import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css";

const BoxList = () => {
	const [ boxes, setBoxes ] = useState([]);
	const addBox = (newBox) => {
		setBoxes((boxes) => [ ...boxes, { ...newBox, id: uuid() } ]);
	};
	const removeBox = (id) => {
		setBoxes((boxes) => boxes.filter((box) => box.id !== id));
	};
	return (
		<div>
			<h3>Colored Boxes</h3>
			<div className="BoxList-form">
				<NewBoxForm addBox={addBox} />
			</div>
			<div className="BoxList">
				{boxes.map(({ id, color, width, height }) => (
					<Box
						key={id}
						id={id}
						backgroundColor={color}
						width={width}
						height={height}
						removeBox={removeBox}
					/>
				))}
			</div>
		</div>
	);
};

export default BoxList;
