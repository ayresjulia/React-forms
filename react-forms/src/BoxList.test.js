import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {
	render(<BoxList />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});

it("should add new box with delete button", () => {
	const { queryByText, getByLabelText } = render(<BoxList />);
	const colorInput = getByLabelText("Color");
	const widthInput = getByLabelText("Width");
	const heightInput = getByLabelText("Height");

	const btn = queryByText("Add");

	expect(queryByText("X")).not.toBeInTheDocument();
	fireEvent.change(colorInput, { target: { value: "red" } });
	fireEvent.change(widthInput, { target: { value: "100" } });
	fireEvent.change(heightInput, { target: { value: "100" } });
	fireEvent.click(btn);
	expect(queryByText("X")).toBeInTheDocument();
});

it("should delete a box", () => {
	const { queryByText, getByLabelText } = render(<BoxList />);
	const colorInput = getByLabelText("Color");
	const widthInput = getByLabelText("Width");
	const heightInput = getByLabelText("Height");

	const btn = queryByText("Add");

	expect(queryByText("X")).not.toBeInTheDocument();
	fireEvent.change(colorInput, { target: { value: "red" } });
	fireEvent.change(widthInput, { target: { value: "100" } });
	fireEvent.change(heightInput, { target: { value: "100" } });
	fireEvent.click(btn);
	expect(queryByText("X")).toBeInTheDocument();
	fireEvent.click(queryByText("X"));
	expect(queryByText("X")).not.toBeInTheDocument();
});
