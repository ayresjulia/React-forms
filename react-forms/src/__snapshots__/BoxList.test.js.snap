// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Colored Boxes
    </h3>
    <div
      class="BoxList-form"
    >
      <form
        class=""
      >
        <div
          class="form-row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="color"
              >
                Color
              </label>
              <input
                class="form-control"
                id="color"
                name="color"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class=""
                for="width"
              >
                Width
              </label>
              <input
                class="form-control"
                id="width"
                name="width"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class=""
                for="height"
              >
                Height
              </label>
              <input
                class="form-control"
                id="height"
                name="height"
                type="text"
                value=""
              />
            </div>
            <button
              class="btn btn-secondary"
            >
              Add
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="BoxList"
    />
  </div>
</DocumentFragment>
`;
